usr/src/uts/common/brand/lx/procfs/lx_prvnops.c

```diff
@@ -4995,6 +5000,34 @@ lxpr_read_sys_kernel_rand_entavl(lxpr_node_t *lxpnp, lxpr_uiobuf_t *uiobuf)
 	lxpr_uiobuf_printf(uiobuf, "%d\n", swrand_stats.ss_entEst);
 }
 
+/* ARGSUSED */
+static void
+lxpr_read_sys_kernel_rand_uuid(lxpr_node_t *lxpnp, lxpr_uiobuf_t *uiobuf)
+{
+	/*
+	 * Each read from this read-only file should return a new 128-bit
+	 * UUID string in the standard UUID format.
+	 */
+	zone_t *zone = LXPTOZ(lxpnp);
+	uint8_t r[16];
+	char uuid[LX_BOOTID_LEN];
+
+	ASSERT(lxpnp->lxpr_type == LXPR_SYS_KERNEL_RAND_UUID);
+	ASSERT(zone->zone_brand == &lx_brand);
+
+	(void) random_get_bytes(r, sizeof (r));
+	r[6] = 0x40 | (r[6] & 0x0f);
+	r[8] = 0x80 | (r[8] & 0x3f);
```

Comment:
How about adding comments here to make it obvious what these lines so?

```suggestion
	/* Set UUID version to 4 (random) */
	r[6] = 0x40 | (r[6] & 0x0f);
	/* Set UUID variant to 1 */
	r[8] = 0x80 | (r[8] & 0x3f);
```